lib/util/StringXor.js

```diff
@@ -0,0 +1,49 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+
+"use strict";
+
+class StringXor {
+	constructor() {
+		this._value = undefined;
+		this._buffer = undefined;
+	}
+
+	add(str) {
+		let buf = this._buffer;
+		let value;
+		if (buf === undefined) {
+			buf = this._buffer = Buffer.from(str, "latin1");
+			this._value = Buffer.from(buf);
+			return;
+		} else if (buf.length !== str.length) {
+			value = this._value;
+			buf = this._buffer = Buffer.from(str, "latin1");
+			if (value.length < buf.length) {
+				this._value = Buffer.allocUnsafe(buf.length);
+				value.copy(this._value);
+				this._value.fill(0, value.length);
+				value = this._value;
+			}
+		} else {
+			value = this._value;
```

Comment:
```suggestion
			buf = this._buffer = Buffer.from(str, "latin1");
			// this is safe to omit this._value assignment, in toString and updateHash need else if (buf !== undefined) 
             // statement
			return;
		} else if (buf.length !== str.length) {
			value = this._value || Buffer.from(buf);
			buf = this._buffer = Buffer.from(str, "latin1");
			if (value.length < buf.length) {
				this._value = Buffer.allocUnsafe(buf.length);
				value.copy(this._value);
				this._value.fill(0, value.length);
				value = this._value;
			}
		} else {
			value = this._value || Buffer.from(buf);
```

improvement for one entry only

Reply:
I measured that but it is slower. I think because of the additional check `this._value || Buffer.from(buf)` in the hot path.

Reply:
sound like copy pasting buffers is pretty fast... probably because runtime already allocated enough memory..